lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -890,7 +887,14 @@ private function persistNew($class, $entity)
             $idValue = $idGen->generate($this->em, $entity);
 
             if ( ! $idGen instanceof \Doctrine\ORM\Id\AssignedGenerator) {
-                $idValue = array($class->identifier[0] => $idValue);
+                $idField = $class->identifier[0];
```

Comment:
I just digged into this conversion in depth, and it seems wrong. The `AssignedGenerator` already uses a PHP value internally, and never produces an SQL value. This code can therefore be dropped, and test code needs to be re-validated